iscsi/iscsi.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2017 The Prometheus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iscsi
+
+import (
+	"path/filepath"
+
+	"github.com/prometheus/procfs/internal/fs"
+)
+
+// TARGETPATH is static sys path for iscsi target
+const TARGETPATH = "/kernel/config/target/iscsi"
+
+// TARGETCORE static sys path for backstore
+const TARGETCORE = "/kernel/config/target/core"
+
+// FS represents the pseudo-filesystem sysfs, which provides an interface to
+// kernel data structures.
+type FS struct {
+	sys *fs.FS
+}
+
+// NewFS returns a new FS mounted under the given mountPoint. It will error
+// if the mount point can't be read.
+func NewFS(mountPoint string) (FS, error) {
+	sysfs, err := fs.NewFS(mountPoint)
```

Comment:
Are you suggesting I should add configfs into the sysfs module? or more like something like below? 

```
// ConfigFsPath is static configfs mount point
const ConfigFsPath= "/kernel/config"

// TargetPath is static sys path for iscsi target
const TargetPath = "/target/iscsi"

// TARGETCORE static sys path for backstore
const TargetCore = "/target/core"

// FS represents the pseudo-filesystem configfs, which provides an interface to
// iscsi kernel data structures.
type FS struct {
    configfs *fs.FS
}

// NewFS returns a new configfs mounted under the given mountPoint. It will error
// if the mount point can't be read.
func NewFS(configfsMountPoint string) (FS, error) {
    if strings.TrimSpace(configfsMountPoint) == "" {
        // /sys/kernel/config
        configfsMountPoint = fs.DefaultSysMountPoint + ConfigFsPath
    }
    sysfs, err := fs.NewFS(configfsMountPoint)
    if err != nil {
        return FS{}, err
    }
    return FS{&configfs}, nil
}
```